### PR TITLE
Allowing netspec to have empty top (ntop=0)

### PR DESCRIPTION
**Pull request by philkr**
Just as it says, allows specification of layers with empty tops (eg Silence, HDF5Output).

### CODE REVIEW

File: python/caffe/net_spec.py [hunk 1]
```diff
@@ -87,6 +87,12 @@ def to_proto(self):
         this top."""
 
         return to_proto(self)
+class NoTop:
```
Comment (shelhamer) [inline python/caffe/net_spec.py]: Add two newlines for pep8itude.

**Comment by shelhamer** [comment]
Looks good to me but @longjon should check.

File: python/caffe/net_spec.py (hunk 1 above)
Comment (philkr) [inline python/caffe/net_spec.py]: hmm that pep8... done.

**Comment by longjon** [comment]
Thanks for PRing this. I have a rather simpler implementation myself, so stay tuned.

**Comment by philkr** [comment]
Simpler than 20 line changes? :)

**Comment by longjon** [comment]
See https://github.com/longjon/caffe/commit/b464a6e5d2e381185bd8aa0b9cbcba53c9819585, and preparations in https://github.com/longjon/caffe/commit/96c2fe1de80c9752b992c4578a3ce46028d21fc5.

**Comment by philkr** [comment]
#2813 seems to be functionally the same, at least for what I'm using it for. So I don't have any objections going with the other solution, even though it's 50 line changes :)

**Comment by shelhamer** [comment]
Closing in favor of #2813. Thanks for getting this rolling @philkr

**Pull request closed without merging**